File: sdks/python/examples/basic.py
"""
Basic example using Realm.ai Python SDK (HTTP client mode)
"""

from realm import RealmClient

def main():
    # Initialize client
    client = RealmClient(
        base_url="http://localhost:8080",
        api_key=None,  # Optional
    )

    try:
        # Simple completion
        response = client.completions(
            prompt="What is the capital of France?",
            max_tokens=50,
            temperature=0.7,
        )

        print("Response:", response.get("text", ""))
        print("Model:", response.get("model", "unknown"))

        # List available models
        models = client.models()
        print("\nAvailable models:")
        for model in models.get("models", []):
            print(f"  - {model.get('id')}: {model.get('name')}")

    except Exception as e:
        print(f"Error: {e}")
    finally:
        client.close()


if __name__ == "__main__":
    main()

File: sdks/js/examples/basic.ts
/**
 * Basic example - Single model usage
 */

import { Realm } from '@realm-ai/sdk';
import * as fs from 'fs';

async function main() {
  // Initialize Realm (single model instance)
  const realm = new Realm();

  // Initialize WASM module
  // In production, WASM should be bundled or provided via wasmPath
  await realm.init(undefined, '../wasm/realm_wasm.js');

  // Load model
  console.log('Loading model...');
  const modelBytes = fs.readFileSync('./models/llama-2-7b.gguf');
  await realm.loadModel(modelBytes, 'llama-7b');
  console.log('✅ Model loaded');

  // Set generation config
  realm.setConfig({
    maxTokens: 50,
    temperature: 0.7,
  });

  // Generate
  const response = await realm.generate('What is the capital of France?');
  console.log('Response:', response.text);
  console.log('Model:', response.model);

  // Cleanup
  realm.dispose();
}

main().catch(console.error);

